subsys/net/lib/app/CMakeLists.txt

```diff
@@ -1,10 +1,10 @@
-#zephyr_library()
```

Comment:
Please squash the original commit and the revert together.

Reply:
> Please squash the original commit and the revert together.

Done. I still left a separate commit for some of cmake changes, so it was clear why they were done, in case it will need further adjustment later (which is likely, as cmake settings for these parts need cleanup).